lib/rules/i18n-no-reference.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+const { I18N_METHODS } = require('../i18n.js');
+
+function isReferenceNode(node) {
+  if (node.type === 'TemplateLiteral') return false;
+
+  return node.type !== 'Literal' || typeof node.value !== 'string'
+}
+
+function extractI18nMethodAndArgs(node) {
+  let result = [];
+
+  if (node.type === 'CallExpression') {
+    const { callee } = node;
+    if (callee.type === 'MemberExpression') {
+      const { object, property } = callee;
+      if (object.name === 'i18n') {
+        result = [property.name, node.arguments];
+      }
+    }
+  }
+
+  return result;
+}
+
+
+module.exports = {
+  meta: {
+    docs: {
+      category: 'Possible Errors',
+      description:
+        'Ensure predictable static values are passed as i18n method arguments',
+      recommended: false,
```

Comment:
If you mark the new rules as `recommended`, they will be automatically used in addons-frontend: https://github.com/mozilla/addons-frontend/blob/8687a2f3d6b4a0366f4b89ec51e160d0a35ffdd0/.eslintrc#L4

Reply:
You could also add a link to the github issue that led to the creation of this rule using: `githubIssue: '<link>'` (which will be used in the generated docs).

Reply:
Done.